public/app/plugins/datasource/prometheus/datasource.test.ts

```diff
@@ -100,6 +100,24 @@ describe('PrometheusDatasource', () => {
         expect(response[0].state).toBe(LoadingState.Done);
       });
     });
+
+    it('throws if using direct access', async () => {
+      const instanceSettings = {
+        url: 'proxied',
+        directUrl: 'direct',
+        user: 'test',
+        password: 'mupp',
+        access: 'direct',
+        jsonData: {
+          customQueryParameters: '',
+        } as any,
+      } as unknown as DataSourceInstanceSettings<PromOptions>;
+      const directDs = new PrometheusDatasource(instanceSettings, templateSrvStub as any, timeSrvStub as any);
+
+      await expect(
+        lastValueFrom(directDs.query(createDataRequest([{}, {}], { app: CoreApp.Dashboard })))
```

Comment:
Can you also repeat this line/test for other methods like `metricFindQuery`, `testDatasource`, `annotationsQuery`, `getTagKeys`, `getTagValues`.

Reply:
Good call, I added tests and `annotationsQuery` wasn't actually covered with the error because it didn't end in the same request method. I could not test `metricFindQuery` because of problems with mocking but I made sure it ended up in the proper request method.